components/BottomAppBar/src/private/MDCBottomAppBarView.m

```diff
@@ -0,0 +1,250 @@
+/*
+ Copyright 2016-present the Material Components for iOS authors. All Rights Reserved.
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+#import "MDCBottomAppBarView.h"
+
+#import "MDCBottomAppBarAttributes.h"
+#import "MDCBottomAppBarLayer.h"
+#import "MaterialRTL.h"
+
+static NSString *kMDCBottomAppBarViewAnimKeyString = @"AnimKey";
+static NSString *kMDCBottomAppBarViewEmptyString = @"";
+static NSString *kMDCBottomAppBarViewPathString = @"path";
+static NSString *kMDCBottomAppBarViewPositionString = @"position";
+
+@interface MDCBottomAppBarCutView : UIView
+
+@end
+
+@implementation MDCBottomAppBarCutView
+
+// Allows touch events to pass through so MDCBottomAppBarController can handle touch events.
+- (UIView *)hitTest:(CGPoint)point withEvent:(UIEvent *)event {
+  UIView *view = [super hitTest:point withEvent:event];
+  return view == self ? nil : view;
+}
+
+@end
+
+@interface MDCBottomAppBarView () <CAAnimationDelegate>
+
+@property(nonatomic, assign) CGFloat bottomBarHeight;
+@property(nonatomic, strong) MDCBottomAppBarCutView *cutView;
+@property(nonatomic, strong) MDCBottomAppBarLayer *bottomBarLayer;
+@property(nonatomic, assign) UIUserInterfaceLayoutDirection layoutDirection;
+
+@end
+
+@implementation MDCBottomAppBarView
+
+- (void)setupBottomBarView {
+  self.cutView = [[MDCBottomAppBarCutView alloc] initWithFrame:self.bounds];
```

Comment:
`cutView` is used with both the `_` and the setter here. Is one better for this method?

Reply:
Changed to self.